src/highdicom/pr/sop.py

```diff
@@ -0,0 +1,1636 @@
+                        ))
+                    lut_id = tuple(lut_info)
+                    by_lut[lut_id].append(ref_im)
+
+            softcopy_voi_luts = []
+            for (width, center, exp, func), im_list in by_window.items():
+                if len(im_list) == len(referenced_images):
+                    # All datasets included, no need to include the referenced
+                    # images explicitly
+                    refs_to_include = None
+                else:
+                    # Include specific references
+                    refs_to_include = ReferencedImageSequence(im_list)
+
+                softcopy_voi_luts.append(
+                    SoftcopyVOILUT(
+                        window_center=center,
+                        window_width=width,
+                        window_explanation=exp,
+                        voi_lut_function=func,
+                        referenced_images=refs_to_include
+                    )
+                )
+
+            for lut_id, im_list in by_lut.items():
+                if len(im_list) == len(referenced_images):
+                    # All datasets included, no need to include the referenced
+                    # images explicitly
+                    refs_to_include = None
+                else:
+                    # Include specific references
+                    refs_to_include = ReferencedImageSequence(im_list)
+
+                luts = [
+                    LUT(
+                        first_mapped_value=fmv,
+                        lut_data=np.frombuffer(
+                            data,
+                            np.uint8 if ba == 8 else np.uint16
+                        ),
+                        lut_explanation=exp
+                    )
+                    for (fmv, ba, exp, data) in lut_id
+                ]
+                softcopy_voi_luts.append(
+                    SoftcopyVOILUT(
+                        referenced_images=refs_to_include,
+                        luts=luts
+                    )
+                )
+
+            if len(softcopy_voi_luts) > 0:
+                self.SoftcopyVOILUTSequence = softcopy_voi_luts
+
+    @staticmethod
+    def _get_icc_profile(ref_im: Dataset) -> Union[bytes, None]:
+        """Get a single ICC profile from a dataset, if one exists.
+
+        Parameters
+        ----------
+        ref_im: pydicom.Dataset
+            Image dataset from which to extract an ICC profile.
+
+        Returns
+        -------
+        Union[bytes, None]:
+            An ICC profile as bytes, if one is found in the image, or None if
+            none can be found.
+
+        Raises
+        ------
+        ValueError:
+            If multiple, non-matching ICC profiles are found in the image.
+
+        """
+        if hasattr(ref_im, 'ICCProfile'):
+            return ref_im.ICCProfile
+        elif hasattr(ref_im, 'OpticalPathSequence'):
+            icc_profile = ref_im.OpticalPathSequence[0].ICCProfile
```

Comment:
@dclunie I realized that there is an issue with the standard that may require a correction.

The Color Softcopy Presentation State IOD allows referencing multiple images, but one can only provide one ICC Profile. There is no guarantee that all referenced images contain the same ICC Profile. For example, in case of slide microscopy, the THUMBNAIL, VOLUME, LABEL, and OVERVIEW images may all be acquired with a different camera. This could lead to problem upon display.